app/rewards/spec.yaml

```diff
@@ -0,0 +1,117 @@
+openapi: 3.0.0
+info:
+  title: "QueueTime Flask REST API"
+  description: "QueueTime API for finding wait times of various points of interests at McMaster University"
+  version: "1.0.0"
+
+components:
+  schemas:
+    RewardEvent:
+      type: object
+      required:
+        - source
+        - points
+        - timestamp
+      properties:
+        source:
+          type: string
+          description: Source of the reward points change
+          example: referral_bonus
+        points:
+          type: integer
+          description: Number of points added or removed
+          example: 25
+        timestamp:
+          type: string
+          format: date-time
+          description: Timestamp of the reward event
+    ErrorResponse:
+      type: object
+      required:
+        - error
+        - message
+      properties:
+        error:
+          type: string
+          description: Error type
+          example: Bad Data
+        message:
+          type: string
+          description: Error description
+          example: "Invalid data provided"
+
+paths:
+  /user/referral/{code}:
+    post:
+      x-openapi-router-controller: app.rewards.api
+      operationId: submit_referral_code
+      tags:
+        - User
+      summary: Submit a referral code
+      description: During signup submit a referral code to earn points as part of a referral bonus
+      parameters:
+        - name: code
+          in: path
+          description: User referral code
+          required: True
+          schema:
+            type: string
+            pattern: "^[A-Z]{6}$"
+          example: AHDBETX
+      responses:
+        "204":
+          description: Valid referral code submitted and accepted
+        "404":
+          description: Referral code not found
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorResponse"
+        "400":
+          description: Referral code invalid
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorResponse"
+      security:
```

Comment:
If we already know we need to secure all the endpoints in rewards, this can also be added globally to the whole file instead of each individual endpoint

Reply:
Updated to add security to all paths except health check.